File: client/java/src/avt/OptionScr.java
package avt;

import java.io.ByteArrayOutputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class OptionScr extends MyScreen {
   public static OptionScr instance;
   private int point = 0;
   private int focus = 0;
   private int max = 5;
   public int[] mapFocus;
   public int volume = 0;
   private int xL;
   private int _hText_;
   private MyScreen lastScr;
   public static boolean isVirTualKey = false;
   public static boolean e = false;
   private boolean[] isPaint;

   public static OptionScr gI() {
      if (instance == null) {
         instance = new OptionScr();
      }

      return instance;
   }

   public final void switchToMe() {
      this.initSize();
      this.lastScr = Canvas.currentMyScreen;
      super.switchToMe();
      this.load();
   }

   public final void commandTab(int var1, int var2) {
      switch (var1) {
         case 0:
            this.save(this.volume);
            this.lastScr.switchToMe();
         default:
      }
   }

   public OptionScr() {
      this.isPaint = new boolean[this.max];
   }

   public final void initSize() {
      super.left = new Command(T.complete, 0);
      this._hText_ = MyScreen.hText;
      this.xL = Canvas.h;
      int var1 = PaintPopup.hTab + (AvMain.hDuBox << 1);
      if (this.isPaint != null) {
         int var2;
         for(var2 = 0; var2 < this.isPaint.length; ++var2) {
            if (this.isPaint[var2]) {
               var1 += this._hText_;
            }
         }

         var2 = 176;
         if (Canvas.w < 176) {
            var2 = Canvas.w;
         }

         PaintPopup.gI().a(T.option, var2 * AvMain.hd, var1, 1);
         if (Canvas.currentMyScreen != this) {
            for(var1 = 0; var1 < 3; ++var1) {
               this.isPaint[var1] = true;
            }

            if (Canvas.E) {
               this.isPaint[3] = true;
            }

            this.mapFocus = new int[this.max];
         }
      }

   }

   public final void save(int var1) {
      this.volume = var1;
      ByteArrayOutputStream var2 = new ByteArrayOutputStream();
      DataOutputStream var3 = new DataOutputStream(var2);

      try {
         var3.writeByte(var1);

         for(int var4 = 0; var4 < this.max; ++var4) {
            var3.writeByte(this.mapFocus[var4]);
         }
      } catch (IOException var6) {
         var6.printStackTrace();
      }

      try {
         CRes.saveRMS("avatarShowName", var2.toByteArray());
         var3.close();
      } catch (Exception var5) {
         var5.printStackTrace();
      }

      this.init();
      SoundManager.a.a(var1 / 10);
   }

   public final void load() {
      this.initSize();
      DataInputStream var1 = AvatarData.loadRMS("avatarShowName");
      isVirTualKey = false;
      if (var1 != null) {
         try {
            this.volume = var1.readByte();
            this.mapFocus = new int[this.max];

            for(int var2 = 0; var2 < this.max; ++var2) {
               this.mapFocus[var2] = var1.readByte();
               if (this.mapFocus[var2] > 1) {
                  this.mapFocus[var2] = 0;
               }
            }

            var1.close();
         } catch (Exception var3) {
            AvatarData.delErrorRms("avatarShowName");
         }

         this.init();
         SoundManager.a.a(this.volume / 10);
      }

   }

   private void init() {
      if (Canvas.E) {
         e = this.mapFocus[3] == 1;
      }

      Canvas.a();
   }

   public final void updateKey() {
      super.updateKey();
      if (Canvas.a(2)) {
         this.setMapFocus_(-1);
      } else if (Canvas.a(8)) {
         this.setMapFocus_(1);
      } else if (Canvas.a(4)) {
         this.setMapFocus(-1);
      } else if (Canvas.a(6)) {
         this.setMapFocus(1);
      }

      if (Canvas.isPointerClick && Canvas.isPointer(PaintPopup.gI().x, PaintPopup.gI().y, PaintPopup.gI().w, PaintPopup.gI().h)) {
         Canvas.isPointerClick = false;
         if (Canvas.isPointer(PaintPopup.gI().x, PaintPopup.gI().y, PaintPopup.gI().w, PaintPopup.gI().h)) {
            int var1;
            for(int var2 = var1 = (Canvas.py - (PaintPopup.gI().y + PaintPopup.hTab + AvMain.hDuBox)) / this._hText_; var2 >= 0; --var2) {
               if (!this.isPaint[var2]) {
                  ++var1;
               }
            }

            if (var1 == this.focus) {
               if (this.mapFocus[this.focus] == 1) {
                  this.setMapFocus(-1);
               } else {
                  this.setMapFocus(1);
               }
            }

            if (var1 >= this.max) {
               var1 = this.max - 1;
            }

            this.focus = var1;
         }
      }

   }

   private void setMapFocus_(int var1) {
      while(true) {
         this.focus += var1;
         if (this.focus < 0) {
            this.focus = this.max - 1;
         }

         if (this.focus >= this.max) {
            this.focus = 0;
         }

         if (this.isPaint[this.focus]) {
            return;
         }

         var1 /= CRes.abs(var1);
      }
   }

   private void setMapFocus(int var1) {
      if (this.focus == 2) {
         this.volume += var1 * 10;
         if (this.volume < 0) {
            this.volume = 100;
         }

         if (this.volume > 100) {
            this.volume = 0;
            return;
         }
      } else {
         if (this.mapFocus[this.focus] == 0) {
            this.mapFocus[this.focus] = 1;
            return;
         }

         this.mapFocus[this.focus] = 0;
      }

   }

   public final void update() {
      this.lastScr.update();
      if (this.xL != 0) {
         this.xL += -this.xL >> 1;
         if (this.xL < 0) {
            this.xL = 0;
         }
      }

   }

   public final void paint(Graphics var1) {
      this.lastScr.paintMain(var1);
      this.paintMain(var1);
      super.paint(var1);
   }

   public final void paintMain(Graphics var1) {
      var1.translate(-var1.getTranslateX(), -var1.getTranslateY());
      var1.translate(0, this.xL);
      PaintPopup.gI().paint(var1);
      var1.translate(Canvas.hw - 65, PaintPopup.gI().y + PaintPopup.hTab + AvMain.hDuBox);
      if (this.point >= 4) {
         this.point = 0;
      }

      int var3 = -AvMain.hNormal / 2 + this._hText_ / 2;
      int var4 = 0;

      for(int var5 = 0; var5 < this.max; ++var5) {
         if (this.isPaint[var5]) {
            Canvas.normalFont.drawString(var1, T.sms[var5][2], -50 * (AvMain.hd - 1), var4 + var3, 0);
            Canvas.normalFont.drawString(var1, T.sms[var5][this.mapFocus[var5]], 52 + 50 * AvMain.hd, var4 + var3 - 1, 2);
            byte var2 = 0;
            int var6;
            if ((var6 = Canvas.normalFont.getWidth(T.sms[var5][this.mapFocus[var5]]) + 10 + 15 * (Canvas.stypeInt + 1) + PaintPopup.b.frameWidth) < 25 * AvMain.hd) {
               var6 = 25 * AvMain.hd;
            }

            if (var5 == this.focus) {
               var2 = 1;
            }

            int var7 = var4 + var3 + AvMain.hNormal / 2 - PaintPopup.b.frameHeight / 2;
            PaintPopup.b.drawFrame(var2, 52 + 50 * AvMain.hd - var6 / 2, var7, 0, var1);
            PaintPopup.b.drawFrame(var2, 52 + 50 * AvMain.hd + var6 / 2 - PaintPopup.b.frameWidth, var7, 2, var1);
            var4 += this._hText_;
         }
      }

      Canvas.normalFont.drawString(var1, String.valueOf(this.volume), 52 + 50 * AvMain.hd, 2 * this._hText_ + var3, 2);
      ++this.point;
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/OptionScr.cs
using System;
using System.IO;

public class OptionScr : MyScreen
{
	public static OptionScr instance;

	private int point;

	private int focus;

	private int max = 5;

	public int[] mapFocus;

	public int volume = 10;

	private int xL;

	private new int hText;

	public const int SHOW_NAME = 0;

	public const int SHOW_DIRECTION = 1;

	public const int SOUND = 2;

	private MyScreen lastScr;

	public static bool isVirTualKey;

	private bool[] isPaint;

	public int x;

	public int y;

	public int w;

	public int h;

	public int hCell;

	private bool isTranKey;

	private new int indexLeft = -1;

	private new int indexRight = -1;

	public OptionScr()
	{
		isPaint = new bool[max];
	}

	public static OptionScr gI()
	{
		if (instance == null)
		{
			instance = new OptionScr();
		}
		return instance;
	}

	public override void switchToMe()
	{
		initSize();
		lastScr = Canvas.currentMyScreen;
		base.switchToMe();
		load();
		mapFocus[2] = 1;
		if (!Main.isCompactDevice)
		{
			volume = 0;
			mapFocus[0] = 0;
			mapFocus[1] = 0;
			mapFocus[2] = 0;
		}
		else if (volume == 0)
		{
			mapFocus[2] = 0;
		}
	}

	public override void commandActionPointer(int index, int subIndex)
	{
		if (index == 0)
		{
			doClose();
		}
	}

	public void initSize()
	{
		center = new Command(T.complete, 0, this);
		hText = MyScreen.hText;
		xL = Canvas.h;
		int num = PaintPopup.hTab + AvMain.hDuBox * 2 - 10;
		if (isPaint != null)
		{
			w = 200 * AvMain.hd;
			h = 200 * AvMain.hd;
			x = (Canvas.w - w) / 2;
			y = (Canvas.hCan - h) / 2;
			hCell = h / 4;
			for (int i = 0; i < 3; i++)
			{
				isPaint[i] = true;
			}
			isPaint[3] = true;
			mapFocus = new int[max];
		}
	}

	protected void doClose()
	{
		save(volume);
		lastScr.switchToMe();
	}

	public void save(int volume)
	{
		this.volume = volume;
		DataOutputStream dataOutputStream = new DataOutputStream();
		try
		{
			dataOutputStream.writeByte((sbyte)volume);
			for (int i = 0; i < max; i++)
			{
				dataOutputStream.writeByte((sbyte)mapFocus[i]);
			}
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
		try
		{
			RMS.saveRMS("avatarShowName", dataOutputStream.toByteArray());
			dataOutputStream.close();
		}
		catch (Exception e2)
		{
			Out.logError(e2);
		}
		init();
	}

	public void load()
	{
		initSize();
		DataInputStream dataInputStream = AvatarData.initLoad("avatarShowName");
		isVirTualKey = false;
		if (dataInputStream == null)
		{
			return;
		}
		try
		{
			volume = dataInputStream.readByte();
			mapFocus = new int[max];
			for (int i = 0; i < max; i++)
			{
				mapFocus[i] = dataInputStream.readByte();
				if (mapFocus[i] > 1)
				{
					mapFocus[i] = 0;
				}
			}
			dataInputStream.close();
		}
		catch (IOException e)
		{
			Out.logError(e);
		}
		init();
	}

	private void init()
	{
		Canvas.paint.setLanguage();
	}

	public override void updateKey()
	{
		base.updateKey();
		if (!Main.isCompactDevice)
		{
			return;
		}
		if (Canvas.isPointerClick)
		{
			int num = hCell;
			for (int i = 0; i < 3; i++)
			{
				if (Canvas.isPoint(x + 90 * AvMain.hd - 20 * AvMain.hd, y + (i + 1) * hCell - 20 * AvMain.hd, 40 * AvMain.hd, 40 * AvMain.hd))
				{
					isTranKey = true;
					indexLeft = i;
					Canvas.isPointerClick = false;
				}
				else if (Canvas.isPoint(x + w - 30 * AvMain.hd - 20 * AvMain.hd, y + (i + 1) * hCell - 20 * AvMain.hd, 40 * AvMain.hd, 40 * AvMain.hd))
				{
					isTranKey = true;
					indexRight = i;
					Canvas.isPointerClick = false;
				}
				num += hCell;
			}
		}
		if (!isTranKey)
		{
			return;
		}
		if (Canvas.isPointerDown)
		{
			if (indexLeft != -1 && !Canvas.isPoint(x + 90 * AvMain.hd - 20 * AvMain.hd, y + (indexLeft + 1) * hCell - 20 * AvMain.hd, 40 * AvMain.hd, 40 * AvMain.hd))
			{
				indexLeft = -1;
			}
			else if (indexRight != -1 && !Canvas.isPoint(x + w - 30 * AvMain.hd - 20 * AvMain.hd, y + (indexRight + 1) * hCell - 20 * AvMain.hd, 40 * AvMain.hd, 40 * AvMain.hd))
			{
				indexRight = -1;
			}
		}
		if (Canvas.isPointerRelease)
		{
			Canvas.isPointerRelease = false;
			isTranKey = false;
			if (indexLeft != -1)
			{
				setMapFocus(-1, indexLeft);
				indexLeft = -1;
			}
			else if (indexRight != -1)
			{
				setMapFocus(1, indexRight);
				indexRight = -1;
			}
		}
	}

	private void setMapFocus(int dir, int i)
	{
		if (i == 2)
		{
			volume += dir * 10;
			if (volume < 0)
			{
				volume = 0;
			}
			if (volume > 100)
			{
				volume = 100;
			}
			SoundManager.setVolume(volume / 100);
		}
		else if (mapFocus[i] == 0)
		{
			mapFocus[i] = 1;
		}
		else
		{
			mapFocus[i] = 0;
		}
	}

	public override void update()
	{
		lastScr.update();
		if (xL != 0)
		{
			xL += -xL >> 1;
			if (xL < 0)
			{
				xL = 0;
			}
		}
	}

	public override void paint(MyGraphics g)
	{
		lastScr.paintMain(g);
		paintMain(g);
		base.paint(g);
		g.setColor(0);
	}

	public override void paintMain(MyGraphics g)
	{
		g.translate(0f - g.getTranslateX(), 0f - g.getTranslateY());
		g.translate(0f, xL);
		Canvas.paint.paintPopupBack(g, x, y + 10 * AvMain.hd, w, h - 20 * AvMain.hd, -1, false);
		g.translate(x, y);
		if (point >= 4)
		{
			point = 0;
		}
		int num = hCell;
		for (int i = 0; i < 3; i++)
		{
			if (isPaint[i])
			{
				g.drawImage(PaintPopup.imgMuiIOS[(indexLeft == i) ? 1 : 0][2], 90 * AvMain.hd, num, 3);
				g.drawImage(PaintPopup.imgMuiIOS[(indexRight == i) ? 1 : 0][3], w - 30 * AvMain.hd, num, 3);
				Canvas.normalFont.drawString(g, T.name[i][2], 10 * AvMain.hd, num - Canvas.normalFont.getHeight() / 2, 0);
				Canvas.normalFont.drawString(g, T.name[i][mapFocus[i]], 90 * AvMain.hd + (w - 30 * AvMain.hd - 90 * AvMain.hd) / 2, num - Canvas.normalFont.getHeight() / 2, 2);
				num += hCell;
			}
		}
		Canvas.normalFont.drawString(g, volume + string.Empty, 90 * AvMain.hd + (w - 30 * AvMain.hd - 90 * AvMain.hd) / 2, num - Canvas.normalFont.getHeight() / 2 - hCell, 2);
		point++;
	}
}
